### PR TITLE
#3601 - Fix bug with brick modal results disappearing

### LINKED ISSUE #3601: Brick Modal - Search results disappear at narrow width
When you make the window narrower, at a certain point the brick results disappear.

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Fixes #3601 

## Discussion

- The trade-off here is the tag list scrolls horizontally at narrower widths. I don't think it's that bad though, and I think it's better than figuring out how to make the icons/labels wrap in a non-ugly way.

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer @twschiller

### CODE REVIEW

**Comment by twschiller** [comment]
This is happening for me:

image

**Review by twschiller — CHANGES_REQUESTED** [review]
See comment on unnecessary scroll based

**Comment by BLoe** [comment]
@twschiller I think the scrollbars should be fixed now, let me know if you're still seeing issues.

File: src/components/brickModal/TagList.module.scss [hunk 1]
```diff
@@ -18,11 +18,13 @@
...
 
 .root {
-  display: flex;
+  display: inline-flex;
   flex-direction: column;
   gap: 0.5rem;
   align-items: stretch;
   padding: $bootstrapPadding 0;
+  width: calc(100% - #{$bootstrapPadding});
+  min-width: 160px;
```
Comment (fregante) [inline src/components/brickModal/TagList.module.scss:27]: I think this is a magic number that only works with the current tag names and your fonts/text rendering. I think `min-content` is the right keyword to use, but I'd double check. 

Once that's set I don’t think you need 100% either, the width can be increased via `flex-grow` or its `display:grid` equivalent.

File: src/components/brickModal/TagList.module.scss [hunk 2]
```diff
@@ -18,11 +18,13 @@
 $bootstrapPadding: 15px;
 
 .root {
-  display: flex;
+  display: inline-flex;
```
Comment (fregante) [inline src/components/brickModal/TagList.module.scss:21]: If the parent is flex or grid, this shouldn't make a difference. `inline-*` elements aren't meant for layout

File: src/components/brickModal/BrickModal.tsx [hunk 3]
```diff
@@ -335,7 +335,7 @@ function ActualModal<T extends IBrick>({
                       <LazyGrid
                         height={height}
                         width={width}
-                        columnWidth={width / RESULT_COLUMN_COUNT}
+                        columnWidth={(width - 15) / RESULT_COLUMN_COUNT} // 15px for the scroll bar width
```
Comment (fregante) [inline src/components/brickModal/BrickModal.tsx:338]: Scrollbars vary across systems: macOS defaults to 0, Windows 17px if I remember correctly. Does this module require a width set in pixels?

File: src/components/brickModal/BrickModal.module.scss [hunk 4]
```diff
@@ -24,7 +24,7 @@ $bootstrapPadding: 15px;
   & > :first-child {
     margin: 20px auto;
     width: 89vw;
-    max-width: 1220px; // Fits 3 "recommended bricks" columns
+    max-width: 1220px;
```
Comment (fregante) [inline src/components/brickModal/BrickModal.module.scss:27]: 👍

**Comment by twschiller** [comment]
@BLoe - working better. One NIT would be the placement of the scroll bar relative to the padding

image

One problem I'm experience when scrolling with the wheel is that the first scroll works, but continuing to trying to move the mouse wheel does nothing. Not sure if it's related to this issue. Or we can create a separate ticket

Re: Windows, let's make sure we have @orliesaurus test this out in a release build

**Review by twschiller — APPROVED** [review]
See other comment

**Comment by fregante** [comment]
> the first scroll works, but continuing to trying to move the mouse wheel does nothing

Probably related to the second point in https://github.com/pixiebrix/pixiebrix-extension/issues/3574

This PR doesn't change the behavior for me in that regard

**Comment by fregante** [comment]
Ooops I misclicked the "Update branch" button sorry 😅

**Comment by fregante** [comment]
The most problematic part is Bootstrap’s layout primitives here. More often than not we're fighting them rather being helpful. Bootstrap layouts are meant for text/documents and not for interfaces. Specifically here they use `flex-wrap` on the `Row` which causes the right "column" to wrap. Interfaces shouldn't wrap.

The problem becomes much less severe by just replacing the `row` class with `d-flex` for example (still not good, but better than "empty")


## Code on `main`

Screen Shot 8

## s/row/d-flex/

Screen Shot 7

**Pull request merged**